### DISCUSSION TITLE
Set State inside Use Effect causing the component to recreating in Build

### DISCUSSION THREAD

**Discussion by marlonelima** [vercel/next.js#38526] [Help]
I'm experiencing a rerender problem inside Next.js. I have the following code:

```
  const [price, setPrice] = useState(0);
  const [volume, setVolume] = useState(0);

  const onChange = (ev: ChangeEvent<HTMLInputElement>) => {
    const { value } = ev.target;

    setVolume(Number(value));
  };

  useEffect(() => {
    console.log('Use Effect Without Deps');
  }, []);

  useEffect(() => {
    setPrice(volume * 50);
  }, [volume]);
```

When I change volume, the useEffect with volume calls the `setPrice`. What really happens is that the component is being recreating as new. The useEffect without deps is being executed again. It only happens in builded app.

Next version: 12.1.6

**Comment by icyJoseph** [comment]
Hi,

And you are super sure this only happens with `next build && next start`?

I'd expect this to happen in development mode because of `strictMode`, but if this is happening in the build version, would you mind sharing more about this component? 

If you have a repository to look at, or a reproduction demo, please do share.
